crates/aptos/src/move_tool/mod.rs

```diff
@@ -1223,130 +1214,134 @@ impl Display for FunctionArgType {
             FunctionArgType::Hex => bcs::to_bytes(
                 &hex::decode(arg).map_err(|err| CliError::UnableToParse("hex", err.to_string()))?,
-            ),
-            FunctionArgType::HexArray => {
-                let mut encoded = vec![];
-                for sub_arg in arg.split(',') {
-                    encoded.push(hex::decode(sub_arg).map_err(|err| {
-                        CliError::UnableToParse(
-                            "hex_array",
-                            format!("Failed to parse hex array: {}", err),
-                        )
-                    })?);
-                }
-                bcs::to_bytes(&encoded)
-            },
-            FunctionArgType::String => bcs::to_bytes(arg),
+            )
+            .map_err(|err| CliError::BCS("arg", err)),
+            FunctionArgType::String => bcs::to_bytes(arg).map_err(|err| CliError::BCS("arg", err)),
             FunctionArgType::U8 => bcs::to_bytes(
                 &u8::from_str(arg).map_err(|err| CliError::UnableToParse("u8", err.to_string()))?,
-            ),
+            )
+            .map_err(|err| CliError::BCS("arg", err)),
             FunctionArgType::U16 => bcs::to_bytes(
                 &u16::from_str(arg)
                     .map_err(|err| CliError::UnableToParse("u16", err.to_string()))?,
-            ),
+            )
+            .map_err(|err| CliError::BCS("arg", err)),
             FunctionArgType::U32 => bcs::to_bytes(
                 &u32::from_str(arg)
                     .map_err(|err| CliError::UnableToParse("u32", err.to_string()))?,
-            ),
+            )
+            .map_err(|err| CliError::BCS("arg", err)),
             FunctionArgType::U64 => bcs::to_bytes(
                 &u64::from_str(arg)
                     .map_err(|err| CliError::UnableToParse("u64", err.to_string()))?,
-            ),
+            )
+            .map_err(|err| CliError::BCS("arg", err)),
             FunctionArgType::U128 => bcs::to_bytes(
                 &u128::from_str(arg)
                     .map_err(|err| CliError::UnableToParse("u128", err.to_string()))?,
-            ),
+            )
+            .map_err(|err| CliError::BCS("arg", err)),
             FunctionArgType::U256 => bcs::to_bytes(
                 &U256::from_str(arg)
                     .map_err(|err| CliError::UnableToParse("u256", err.to_string()))?,
-            ),
+            )
+            .map_err(|err| CliError::BCS("arg", err)),
             FunctionArgType::Raw => {
-                let raw = hex::decode(arg)
-                    .map_err(|err| CliError::UnableToParse("raw", err.to_string()))?;
-                Ok(raw)
+                hex::decode(arg).map_err(|err| CliError::UnableToParse("raw", err.to_string()))
+            },
+        }
+    }
+
+    /// Recursively parse argument JSON into BCS representation.
+    pub fn parse_arg_json(&self, arg: &serde_json::Value) -> CliTypedResult<ArgWithType> {
```

Comment:
nit: Would be great if we can unit test and safeguard the parser

Reply:
Agreed. As I mention in https://github.com/aptos-labs/aptos-core/pull/7790#discussion_r1178408642 I'm working on a separate PR that will extend support to JSON input files as well. Perhaps once both input types are supported an end-to-end test will be in order?